src/test/java/com/uwetrottmann/tmdb2/TmdbTestSuite.java

```diff
@@ -0,0 +1,218 @@
+
+import com.googlecode.junittoolbox.SuiteClasses;
+import com.googlecode.junittoolbox.WildcardPatternSuite;
+import com.uwetrottmann.tmdb2.entities.Account;
+import com.uwetrottmann.tmdb2.entities.BaseMovie;
+import com.uwetrottmann.tmdb2.entities.FavoriteMedia;
+import com.uwetrottmann.tmdb2.entities.ListCreateRequest;
+import com.uwetrottmann.tmdb2.entities.ListCreateResponse;
+import com.uwetrottmann.tmdb2.entities.ListOperation;
+import com.uwetrottmann.tmdb2.entities.RatingObject;
+import com.uwetrottmann.tmdb2.entities.WatchlistMedia;
+import com.uwetrottmann.tmdb2.enumerations.AuthenticationType;
+import com.uwetrottmann.tmdb2.enumerations.MediaType;
+import org.assertj.core.util.Strings;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.rules.ExternalResource;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.ArrayList;
+
+import static com.uwetrottmann.tmdb2.TestData.initializeTestingIntegrityData;
+import static com.uwetrottmann.tmdb2.TestData.testList;
+import static com.uwetrottmann.tmdb2.TestData.testListMovies;
+import static com.uwetrottmann.tmdb2.TestData.testMovie;
+import static com.uwetrottmann.tmdb2.TestData.testTvEpisode;
+import static com.uwetrottmann.tmdb2.TestData.testTvSeason;
+import static com.uwetrottmann.tmdb2.TestData.testTvShow;
+
+@RunWith(WildcardPatternSuite.class)
+@SuiteClasses({"**/*Test.class"})
+public class TmdbTestSuite {
+
+    private static String API_KEY;
+    public static String USERNAME ;
+    public static String PASSWORD;
+    static final boolean DEBUG = true;
+
+    static Tmdb unauthenticatedInstance;
+    static Tmdb authenticatedInstance;
+
+    static Boolean initialized = false;
+
+    public static Boolean suiteRunning = false;
+    @ClassRule
+    public static ExternalResource triggerSuiteRunning = new ExternalResource() {
+        @Override
+        protected void before() throws Throwable {
+            suiteRunning = true;
+        }
+    };
+
+    @BeforeClass
+    public static void initializeEnvironment() throws IOException, ParseException {
+        if (initialized)
+            return;
+
+        initialized = true;
+
+        if (Strings.isNullOrEmpty(USERNAME) || Strings.isNullOrEmpty(PASSWORD)) {
+            USERNAME = System.getenv("TMDB_USERNAME");
+            PASSWORD = System.getenv("TMDB_PASSWORD");
+        }
+        if (Strings.isNullOrEmpty(API_KEY)) {
+            API_KEY = System.getenv("TMDB_APIKEY");
+        }
+        if (API_KEY == null) {
+            System.err.println("Your Environment is not Set for Testing. Please Define the proper Environment variables and try again.");
+            System.exit(0);
+        }
+
+        unauthenticatedInstance = new BaseTestCase.TestTmdb(API_KEY);
+        authenticatedInstance = new BaseTestCase.TestTmdb(API_KEY);
+
+        initializeTestingIntegrityData();
+        initializeAccountData();
```

Comment:
I guess you did somewhat understand as you added annotations for account/guest session tests. Maybe a better solution: add a separate test suite for classes that require account setup. May need to split up test classes.

As for the initialize and rollback methods: those should only run for the affected methods/classes, not for all of them.